Sources/swift-build-prebuilts/BuildPrebuilts.swift

```diff
@@ -0,0 +1,331 @@
+                        name: library.name,
+                        products: library.products,
+                        cModules: library.cModules,
+                        artifacts: newArtifacts
+                    )
+                    newLibraries.insert(newLibrary)
+
+                    try await shell("git reset --hard")
+                }
+
+                if let oldManifest = try await downloadManifest(version: version) {
+                    // Add in elements from the old manifest we haven't generated
+                    for library in oldManifest.libraries {
+                        if var newLibrary = newLibraries[library.name] {
+                            var newArtifacts = IdentifiableSet<Workspace.PrebuiltsManifest.Library.Artifact>(newLibrary.artifacts)
+                            for oldArtifact in library.artifacts {
+                                if !newArtifacts.contains(id: oldArtifact.id) {
+                                    newArtifacts.insert(oldArtifact)
+                                }
+                            }
+                            newLibrary.artifacts = .init(newArtifacts.values)
+                            newLibraries.insert(newLibrary)
+                        } else {
+                            newLibraries.insert(library)
+                        }
+                    }
+                }
+                let newManifest = Workspace.PrebuiltsManifest(libraries: .init(newLibraries.values))
+
+                let encoder = JSONEncoder()
+                encoder.outputFormatting = .prettyPrinted
+                let manifestData = try encoder.encode(newManifest)
+                let manifestFile = stageDir.appending("\(swiftVersion)-manifest.json")
+                try manifestData.write(to: manifestFile.asURL)
+            }
+        }
+    }
+
+    func canBuild(_ platform: Workspace.PrebuiltsManifest.Platform) -> Bool {
+        if dockerOnly {
+            return platform.os == .linux
+        }
+#if os(macOS)
+        if platform.os == .macos {
+            return true
+        }
+#elseif os(Windows)
+        if platform.os == .windows {
+            return true
+        }
+#endif
+        return docker && platform.os == .linux
+    }
+
+    func shell(_ command: String) async throws {
+#if os(Windows)
+        let arguments = ["C:\\Windows\\System32\\cmd.exe", "/c", command]
+#else
+        let arguments = ["/bin/bash", "-c", command]
+#endif
+        let process = AsyncProcess(
+            arguments: arguments,
+            outputRedirection: .none
+        )
+        print("Running:", command)
+        try process.launch()
+        let result = try await process.waitUntilExit()
+        switch result.exitStatus {
+        case .terminated(code: let code):
+            if code != 0 {
+                throw StringError("Command exited with code \(code): \(command)")
+            }
+#if os(Windows)
+        case .abnormal(exception: let exception):
```

Comment:
chore (possibly-blocking): Although we don't run tests on windows today, can we add a test case that will exercise this path, which we can then ensure it passes when addressing #8121 ?

Reply:
This is the prebuilts build script. There are no automated tests for it.